packages/itwinui-react/src/core/ExpandableBlock/ExpandableBlock.tsx

```diff
@@ -63,100 +64,254 @@ type ExpandableBlockProps = {
+        </>
       )}
-      style={style}
-      ref={ref}
+    </Box>
+  );
+}) as PolymorphicForwardRefComponent<'button', ExpandableBlockHeaderOwnProps>;
+ExpandableBlockHeader.displayName = 'ExpandableBlock.Header';
+
+// ----------------------------------------------------------------------------
+// ExpandableBlock.ExpandIcon component
+
+const ExpandableBlockExpandIcon = React.forwardRef((props, forwardedRef) => {
+  const { className, children, ...rest } = props;
+  return (
+    <Box
+      className={cx('iui-expandable-block-icon', className)}
+      ref={forwardedRef}
+      {...rest}
+    >
+      {children ?? <SvgChevronRight aria-hidden />}
+    </Box>
+  );
+}) as PolymorphicForwardRefComponent<'div'>;
+ExpandableBlockExpandIcon.displayName = 'ExpandableBlock.ExpandIcon';
+
+// ----------------------------------------------------------------------------
+// ExpandableBlock.LabelArea component
+
+const ExpandableBlockLabelArea = polymorphic.span('iui-expandable-block-label');
+ExpandableBlockLabelArea.displayName = 'ExpandableBlock.LabelArea';
+
+// ----------------------------------------------------------------------------
+// ExpandableBlock.Title component
+
+const ExpandableBlockTitle = React.forwardRef((props, forwardedRef) => {
+  const { className, children, ...rest } = props;
+  return (
+    <Box
+      className={cx('iui-expandable-block-title', className)}
+      ref={forwardedRef}
       {...rest}
     >
+      {children}
+    </Box>
+  );
+}) as PolymorphicForwardRefComponent<'div'>;
+ExpandableBlockTitle.displayName = 'ExpandableBlock.Title';
+
+// ----------------------------------------------------------------------------
+// ExpandableBlock.Caption component
+
+const ExpandableBlockCaption = polymorphic('iui-expandable-block-caption');
+ExpandableBlockCaption.displayName = 'ExpandableBlock.Caption';
+
+// ----------------------------------------------------------------------------
+// ExpandableBlock.EndIcon component
+
+const ExpandableBlockEndIcon = React.forwardRef((props, forwardedRef) => {
+  const { children, ...rest } = props;
+  const { status } = useSafeContext(ExpandableBlockContext);
+
+  const icon = children ?? (status && StatusIconMap[status]());
+  return (
+    <Icon fill={status} ref={forwardedRef} {...rest}>
+      {icon}
+    </Icon>
+  );
+}) as PolymorphicForwardRefComponent<
+  'span',
+  React.ComponentPropsWithoutRef<typeof Icon>
+>;
+ExpandableBlockEndIcon.displayName = 'ExpandableBlock.EndIcon';
+
+// ----------------------------------------------------------------------------
+// ExpandableBlock.Content component
+type ExpandableBlockContentOwnProps = {
+  innerProps?: React.ComponentPropsWithoutRef<'div'>;
+};
+
+const ExpandableBlockContent = React.forwardRef((props, forwardedRef) => {
+  const { className, children, innerProps, ...rest } = props;
+  const { isExpanded } = useSafeContext(ExpandableBlockContext);
+
+  return (
+    <WithCSSTransition in={isExpanded}>
```

Comment:
Tangential to API changes: looks like animation is not working, because the `WithCSSTransition` component doesn't take #1302 into consideration. Might be time to use [pure CSS transitions](https://css-tricks.com/css-grid-can-do-auto-height-transitions/).

It can be done in a separate PR but you might want to start working on that PR *now* in case it requires any API/dom structure changes. You can create a chained PR by creating a branch on top of `expandable-block-composition` and using it as the base branch for your PR.